src/software/ai/navigator/path_planner/path_planner.h

```diff
@@ -15,8 +17,22 @@
 class PathPlanner
 {
    public:
-    virtual std::optional<std::vector<Point>> findPath(const Point &start,
-                                                       const Point &dest) = 0;
+    /**
+     * Returns a "good" path between start and dest.
+     *
+     * @param start start point
+     * @param dest destination point
+     * @param field field
+     * @param obstacles obstacles to avoid
+     *
+     * @return the optimal path between start and dest
+     * 		if no valid path then return std::monostate
+     * 		if spline navigator then return a Curve
+     * 		if point navigator then return a std::vector of points
+     */
+    virtual std::variant<std::monostate, Curve, std::vector<Point>> findPath(
```

Comment:
I strongly disagree with having a `std::monostate` in this. Returning an empty path is fine, but I don't think there's any good reason where not only do we not want to return a path, we don't even want to indicate the type that we're returning.

Reply:
I propose `std::variant<std::optional<std::vector<Point>>, std::optional<Curve>>` then. Actually what if want a series of curves to represent a spline. Then we can have `std::variant<std::optional<std::vector<Point>>, std::optional<std::vector<Curve>>>` !!

Reply:
I don't see why we need the `std::optional`s here. I don't think this is a case where `std::optional` adds value by being able to express an "empty" return value, since `std::vector` can already do that. It just causes more checking to be necessary.

Reply:
🤔good point I wonder why we ever used optional then

Reply:
It should definitely be `std::variant<std::vector<Point>, std::vector<Curve>>` though (rather then just one curve)!

Reply:
Done